openedx/core/djangoapps/course_roles/docs/decisions/0001-course_level_roles.rst

```diff
@@ -0,0 +1,89 @@
+1. Course Level Roles
+######################
+
+
+Status
+******
+
+**Provisional** *2023-09-13*
+
+The status will be updated to *Accepted* upong completion of reimplementation. Related work:
+
+- `Course Roles Setup <https://2u-internal.atlassian.net/browse/ROLES-2>`
+
+
+Context
+*******
+
+There is currently no single functionality being used by the majority of the codebases that are a part of Open edX
+that allows for adding roles (permission sets) for a user or group of users at the course and organization level
+that provides the flexibility to allow users to edit or create new roles (permission sets) that can be assigned to users or groups of users.
+
+
+Decision
+********
+
+Modified LMS table and solution - within LMS/CMS
+------------------------------------------------
+- We will add new tables to manage roles, permission sets, and groups of users that are assigned to roles.
+- We will add additional tables to manage role permissions (to accommodate custom roles).
+- We will create the new tables within the current DB schema and LMS/CMS repo.
+
+
+Consequences
+************
+
+- New tables diagram: `<https://whimsical.com/roles-and-permissions-project-plan-WrBuVdYt9D3e3PHkbr4h7P>``
```

Comment:
Can we embed the table design diagrams that @julianpalmerio has included in the PR for the table migrations? The diagram in whimisical is a possible end state, but not being implemented as part of the initial steps.